modules/apps/object/object-service/src/main/java/com/liferay/object/internal/deployer/ObjectDefinitionDeployerImpl.java

```diff
@@ -375,6 +380,9 @@ public List<ServiceRegistration<?>> deploy(
 		catch (PortalException portalException) {
 			return ReflectionUtil.throwException(portalException);
 		}
+		finally {
+			CompanyThreadLocal.setCompanyId(oldCompanyId);
+		}
```

Comment:
FWIW, it's a little more ergonomic to use `CompanyThreadLocal.setWithSafeCloseable` since it will handle this re-setting logic for you. Here's an [example](https://github.com/drewbrokke/liferay-portal/blob/master/modules/apps/portal-security-sso/portal-security-sso-openid-connect-impl/src/main/java/com/liferay/portal/security/sso/openid/connect/internal/configuration/admin/service/OpenIdConnectProviderManagedServiceFactory.java?plain=1#L92).

Reply:
Though it does seem like you'll have to be careful not to alter the `try/catch` logic at all.